api/client/trust.go

```diff
@@ -375,34 +375,52 @@ func (cli *DockerCli) trustedPush(repoInfo *registry.RepositoryInfo, tag string,
 
 	defer responseBody.Close()
 
-	targets := []target{}
+	// If it is a trusted push we would like to find the target entry which match the
+	// tag provided in the function and then do an AddTarget later.
+	target := &client.Target{}
+	// Count the times of calling for handleTarget,
+	// if it is called more that once, that should be considered an error in a trusted push.
+	cnt := 0
 	handleTarget := func(aux *json.RawMessage) {
+		cnt++
```

Comment:
All this function does is mutate `target`. It doesn't really matter if it changes it more than once, because `trustedPush` will never use `target` if `cnt > 1` (see line 414). BTW, this check was added in response to one of my earlier comments.

But sure, let's add a similar check inside `handleTarget`. I think that makes the intent clearer.